### ISSUE TITLE
Aria roles on Select input do not conform to WAI-ARIA standards

### ISSUE THREAD

**Issue by phinch** [grommet/grommet#5897]
### Expected Behavior

The Select input, which is built using non-native HTML, should conform to the [WAI-ARIA standards for a collapsible listbox](https://www.w3.org/TR/wai-aria-practices-1.1/examples/listbox/listbox-collapsible.html). This includes having the proper role information indicating that the button will open a listbox when pressed, a role of `listbox` for the element that contains the options, and a role of `option` for each option.

### Actual Behavior

The Select input button does not indicate that a listbox will open when pressed; and the listbox that opens is read by the screenreader as a menu rather than a listbox. These issues have caused issues with Grommet adoption for our team because we are required to meet certain accessibility standards.

### URL, screen shot, or Codepen exhibiting the issue

The behavior can be reproduced on the simple Select input [in the Grommet docs](https://v2.grommet.io/select)

### Steps to Reproduce

1. Use a screenreader to access the Select input and observe the issues reported above

### Your Environment

- Grommet version: 2.19.0

**Comment by jcfilben** [comment]
Hi @phinch I looked into this a bit and it seems like select and listbox, although similar, are two separate things. 

> The listbox role is used for lists from which a user may select one or more items which are static and, unlike HTML select elements, may contain images

More info about listbox: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role

Looking at the [native HTML select](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) the accessibility is behaving in the same way the Grommet Select is. 

Both the HTML select and Grommet select are read as "menu" and the options are read as "menu item". Since select and listbox are two separate things I don't think we should change the Grommet Select to follow the accessibility standards for a listbox and instead we should leave as is since the Grommet Select is currently behaving the same way and HTML select is with regards to accessibility.

**Comment by jcfilben** [comment]
Hi @phinch I am going to go ahead and close this since select and listbox are two separate things but feel free to reopen if you would like to continue the discussion

**Comment by phinch** [comment]
hi @jcfilben! thanks for looking into this. admittedly, i'm not very experienced with accessibility terms or work, so i'm learning as i go here :) I think you're right that I misdiagnosed the issue by saying it was about roles attached to the element, rather than about the interactive behavior and screenreader readout of each element. i do think i've also been using the listbox term incorrectly - perhaps i should have specific "single-select listbox," which i see occasionally referred to in the w3 documentation, including in the example link above. In any case, I'll refrain from using that term again for now, since it does indeed seem to be a different set of interactions 😅 

That all aside, I feel like i'm seeing some differing behavior between the native HTML select and the Grommet Select. When I focus on the Select input, I get the "menu popup collapsed" message from the screenreader, whereas the Grommet Select button just says "Open Drop"; if i remain on the button and let the screenreader continue reading, on the native select it provides instructions for how to show a list of options, whereas in the grommet select it only gives instructions on how to click a button. It also seems like to navigate the options of the native select with a screenreader, I use the up and down arrow keys; and to navigate the options of the grommet select while the screenreader is on, i need to use the left and right arrow keys (it's unclear to me why the up and down doesn't work, but that's what happens at least on my Mac with Voiceover on...)

We've been told by our team's accessibility auditor that these differences make the component prohibitively confusing for screenreader users to understand as a typical select input. I can definitely understand how something like the up-down vs. left-right arrows would be confusing for a screenreader user especially. Do you think these issues would bear addressing? If so, let me know if I should open a separate issue for this, since I kind of botched the initial reporting on this one lol.

Initial readout of select input on focus
![image](https://user-images.githubusercontent.com/9158038/150886265-76bf12e7-18d7-40c5-9132-ef170947f1b6.png)
![image](https://user-images.githubusercontent.com/9158038/150886284-160eee8f-4169-47ac-837f-2b4965cafef6.png)

Additional info as screenreader continues while focused on element
![image](https://user-images.githubusercontent.com/9158038/150886532-a733a621-eb94-4719-a3d7-5ee8e9df6b80.png)
![image](https://user-images.githubusercontent.com/9158038/150886567-1dd9340b-52bc-423e-bc06-76e5d778ae24.png)

**Comment by jcfilben** [comment]
Hi @phinch I am learning as I go as well, so no worries there. Thanks for the detailed explanation and sharing screenshots to show the issue, that is very helpful! I see what you mean about the differences between how the screenreader reads the html select and the grommet select. If you could open a separate issue for this that would be great!
